doc/CounterChecks.markdown

```diff
@@ -0,0 +1,224 @@
+* every time the config is loaded (Tally Server), or
+* at the start of every round (Share Keeper, Data Collector).
+
+If these checks fail, the configuration is rejected and no collection occurs.
+
+### Counters
+
+Each counter counts events in a set of bins.
+
+* Each counter name must be in the list of valid counter names  
+  (This list can have counters added to it at runtime)
+* Counters for bins and sigmas must match
+
+### Noise
+
+Each counter has noise added to it when it is initialised by the data
+collector.
+
+#### Noise Weights
+
+The noise weight reflects how much a client's activity is expected to change
+the counter.
+
+* Each noise weight is a non-negative decimal value
+* The total noise weight is less than the maximum counter value
+
+#### Sigmas
+
+The sigma determines how much noise is added to a counter.
+
+* Sigma counters must have a sigma element
+* Each sigma is a non-negative decimal value
+
+### Bins
+
+The bins determine how events are aggregated in the counter.
+
+* Bin counters must have a bins element
+* Bins have lower and upper bounds which are decimal values  
+  (including positive and negative infinity)
+* The upper bound is strictly greater than the lower bound
+* The bins in a counter must not overlap (but there may be gaps)
+
+### Periods
+
+The periods determine how often nodes communicate, and how long rounds last.
+
+#### Collect Period
+
+Data Collectors collect events for collect period seconds.
+
+```
+tally_server:
+  collect_period: 86400
+```
+
+* the collect period must be a non-negative integer
+* a collect period shorter than 4 seconds does not work reliably
+
+#### Event Period
+
+The tally server checks for new events every event period.
+
+```
+tally_server:
+  event_period: 600
+```
+
+* the event period must be a non-negative integer
+* there must be at least two event periods in each collect period
+* event periods shorter than 2 seconds do not work reliably
+
+#### Checkin Period
+
+Data Collectors and Share Keepers check in every checkin period.
+
+```
+tally_server:
+  checkin_period: 60
+```
+
+* the checkin period must be a non-negative integer
+* the checkin period should be less than or equal to the event period  
+  (checkin periods longer than the event period cause client status update
+  delays)
+* there must be at least two checkin periods in each collect period
+* checkin periods shorter than 2 seconds do not work reliably
+
+#### Delay Period
+
+If the noise allocation changes, the Share Keepers enforce a delay period
+between consecutive rounds.
+
+```
+tally_server:
+  delay_period: 86400
+data_collector:
+  delay_period: 86400
+share_keeper:
+  delay_period: 86400
+```
+
+* the delay period must be a non-negative integer
+* PrivCount warns if the delay period is very short
+
+### Node Thresholds
+
+The node thresholds determine when rounds are run.
+The Tally Server does not start the round until both node thresholds are met.
```

Comment:
i.e., when the number of nodes of each type that have joined our deployment exceeds the configured thresholds.